### PR TITLE
Introduce PreparE

### PR DESCRIPTION
**Pull request by fpagnoux**
Prestation partagée d'éducation de l'enfant
Connected to https://github.com/sgmap/mes-aides-ui/issues/61

### CODE REVIEW

File: openfisca_france/model/prestations/prestations_familiales/paje.py [hunk 1]
```diff
@@ -236,6 +236,13 @@ def function(self, simulation, period):
                                 partiel2 * P.paje.clca.avecab_tx_partiel2))
         return period, paje_clca
 
+reference_input_variable(
+   name ='paje_prepare',
+   column = FloatCol,
+   entity_class = Familles,
+   label = u"Prestation partagée d’éducation de l’enfant",
```
Comment (benjello) [inline openfisca_france/model/prestations/prestations_familiales/paje.py]: Please introduce the acronym PreParE so people can get the origin of the variable name.

**Comment by fpagnoux** [comment]
Done

File: openfisca_france/model/prestations/minima_sociaux/cmu.py [hunk 2]
```diff
@@ -275,7 +276,7 @@ def function(self, simulation, period):
 
         res = cmu_br_i_par[CHEF] + cmu_br_i_par[PART] + forfait_logement
 
-        res += 12 * (aspa + ass + asi + af + cf + asf + paje_clca)
+        res += 12 * (aspa + ass + asi + af + cf + asf + paje_clca + paje_prepare)
```
Comment (benjello) [inline openfisca_france/model/prestations/minima_sociaux/cmu.py]: I know this is not the subject of your commit but I am concerned by the number 12 here. Is it by law or is it a left over from the times we did not deal with infra annual values ?

Comment (fpagnoux) [inline openfisca_france/model/prestations/minima_sociaux/cmu.py]: The resources ceiling for CMU is annual, while those aids are calculated by month. Hence the multiplication by 12, so yes I guess it's a left over. I'll fix it.

Comment (benjello) [inline openfisca_france/model/prestations/minima_sociaux/cmu.py]: Thx

Comment (fpagnoux) [inline openfisca_france/model/prestations/minima_sociaux/cmu.py]: Actually I don't see how to do it using something cleaner. 
@benjello, maybe I can call you to discuss it ?

Comment (benjello) [inline openfisca_france/model/prestations/minima_sociaux/cmu.py]: Yep whenever you want

Comment (fpagnoux) [inline openfisca_france/model/prestations/minima_sociaux/cmu.py]: Thanks, i'll call you tomorrow

Florian

2015-05-06 17:47 GMT+02:00 Mahdi Ben Jelloul <notifications@github.com>:

**Comment by fpagnoux** [comment]
Sorry we didn't manage to talk about it today, we'll have to postpone this fix to the next week :s

**Comment by benjello** [comment]
I was not available when you called. On my side it can wait until next week
;-)

Mahdi Ben Jelloul

**Comment by fpagnoux** [comment]
Hi @benjello,

I tried to get rid of this 12, but I'm not sure there there is a simple way to do it cleanly. 
If you look in the cmu.py, in `cmu_base_ressources_i`, annual periods are already used for all the inputs. What you pointed out in `cmu_base_ressources` is a bit different : it is not inputs, but aids we previously calculated. 
Since in mes-aides, computed aids are always defined for the running month, there value is non-null only for this period. But we are making the hypothesis (all over mes aides) that any aid calculated has been perceived for the last 12 months. We thus want to multiply the monthly value by 12 to get the annual value.  
I looked at the different types of `base_function` in openfisca-core, but I didn't see anything that fits this behavior : take the most recent value, and assume it's always been the same in the past. Maybe I missed something ? 
Otherwise either we need to add a new `base_function`, or we stay with this 12 factor. 

You can call me whenever  suits you to discuss it in more details ;)

**Comment by benjello** [comment]
@fpagnoux you should talk to @eraviart about base_function, he is more expert than I am on that side. Another solution is to create a reform to create your own cmu_bases_ressources wich deals with the idiosyncrasies of mes-adies. Hope it helps.

**Comment by fpagnoux** [comment]
Ok, i'll talk to @eraviart to see how we can figure this out. 
Since this issue is not directly related to the subject of the PR, is it possible to merge this anyway and to deal with it independently ? 

Thanks

**Comment by cbenz** [comment]
I just tried to merge on openfisca/next, one test does not pass:

ERROR: openfisca_france.tests.reforms.test_cesthra.test_cesthra_invalidee

**Comment by cbenz** [comment]
Perhaps you need to add the tests from Ludwig to this PR?

I can do it for all the PR #241 #242 #244 if you'd like

**Comment by fpagnoux** [comment]
I ran the test on the branch, that should be good to merge. 
Are you sure you got the error `ERROR: openfisca_france.tests.reforms.test_cesthra.test_cesthra_invalidee` on this branch ? I think it's related to #242

**Comment by cbenz** [comment]
Oops... you're right it's OK

**Pull request merged**